File: backend/src/main/java/com/poojithairosha/vristopos/service/CompanyService.java
package com.poojithairosha.vristopos.service;

import com.poojithairosha.vristopos.dto.ClientResponse;
import com.poojithairosha.vristopos.dto.CompanyDTO;
import com.poojithairosha.vristopos.model.supplier.Company;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;

import java.util.List;

public interface CompanyService {

    Page<CompanyDTO> searchCompanies(int page, int size, String text);

    List<CompanyDTO> getAllCompanies();

    ClientResponse registerCompany(CompanyDTO companyDTO);

    CompanyDTO getCompanyById(Long id);

    ClientResponse updateCompany(CompanyDTO companyDTO);

}

File: frontend/src/app/user-dashboard/suppliers/companies/company.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {shareReplay} from "rxjs";
import {CompanyDTO} from "../../../dto/CompanyDTO";
import {ClientResponse} from "../../../dto/ClientResponse";

@Injectable({
    providedIn: 'root'
})
export class CompanyService {

    constructor(private http: HttpClient) {
    }

    searchCompanies(text: string, page: number, size: number) {
        return this.http.get<any>(`http://localhost:8080/api/companies/search?text=${text}&page=${page}&size=${size}`).pipe(
            shareReplay(1)
        );
    }

    registerCompany(company: { name: string, email: string, mobile: string, address: string }) {
        return this.http.post<ClientResponse>(`http://localhost:8080/api/companies`, company)
    }

    getCompany(companyId: string) {
        return this.http.get<CompanyDTO>(`http://localhost:8080/api/companies/${companyId}`)
    }

    updateCompany(company: CompanyDTO) {
        return this.http.put<ClientResponse>(`http://localhost:8080/api/companies`, company)
    }
}
